dialer.go

```diff
@@ -219,21 +220,23 @@ func (d *Dialer) Dial(ctx context.Context, instance string, opts ...DialOption)
 	}
 	latency := time.Since(startTime).Milliseconds()
 	go func() {
-		trace.RecordDialLatency(ctx, instance, d.dialerID, latency)
-		trace.RecordConnectionOpen(ctx, instance, d.dialerID)
+		n := atomic.AddUint64(&i.OpenConns, 1)
+		d.metrics.RecordOpenConnections(ctx, int64(n), d.dialerID, i.String())
+		d.metrics.RecordDialLatency(ctx, instance, d.dialerID, latency)
 	}()
 
-	return newInstrumentedConn(tlsConn, instance, d.dialerID), nil
+	return newInstrumentedConn(tlsConn, func() {
+		n := atomic.AddUint64(&i.OpenConns, ^uint64(0))
+		d.metrics.RecordOpenConnections(ctx, int64(n), d.dialerID, i.String())
```

Comment:
Probably not in case the context was canceled. OpenCensus uses the context to associate traces. Here since we're only concerned with a metric, `context.Background()` is probably acceptable.

Reply:
aren't we handing `ctx` into `newInstrumentedConn` here though?